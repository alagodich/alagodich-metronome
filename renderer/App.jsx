import MainWindow from './components/MainWindow.jsx';
import React from 'react';
import ReactDom from 'react-dom';

window.onload = () => {
    const container = document.querySelector('.app');
    if (!container) {
        return;
    }
    ReactDom.render(<MainWindow />, container);
};
