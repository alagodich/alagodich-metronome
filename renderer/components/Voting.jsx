import React, {PureComponent, PropTypes} from 'react';

const propTypes = {
    pair: PropTypes.object,
    vote: PropTypes.func,
    hasVoted: PropTypes.string,
    winner: PropTypes.string
};

class Voting extends PureComponent {
    getPair() {
        return this.props.pair || [];
    }
    isDisabled() {
        return !!this.props.hasVoted;
    }
    hasVotedFor(entry) {
        return this.props.hasVoted === entry;
    }
    hasWinner() {
        return !!this.props.winner;
    }
    render() {
        return (
            <div className="voting">
                {
                    this.hasWinner()
                        ? <span>{this.props.winner}{' winner!'}</span>
                        : (
                            <div className="voting ui buttons">
                                {this.getPair().map(entry => (
                                    <button className="ui button" key={entry} onClick={this.props.vote(entry)}>
                                        {this.hasVotedFor(entry)
                                            ? <i className="heart icon" />
                                            : null
                                        }
                                        {entry}
                                    </button>
                                ))}
                            </div>
                        )
                }
            </div>
        );
    }
}

Voting.propTypes = propTypes;

export default Voting;
