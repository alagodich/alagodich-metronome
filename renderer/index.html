<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Metronome</title>
    <!-- CSS -->
    <link href="../dist/app.css" rel="stylesheet">
  </head>

  <body>
      <div class="ui main container basic segment">
          <div class="app"></div>
      </div>
  </body>

  <script>
    require('../dist/vendors.js');
    require('../dist/app.js');
  </script>

</html>
