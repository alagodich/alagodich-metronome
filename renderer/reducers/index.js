import {combineReducers} from 'redux';
import {routerReducer as routing} from 'react-router-redux';
import counter from './counter';
import voting from './voting';

const rootReducer = combineReducers({counter, voting, routing});

export default rootReducer;
