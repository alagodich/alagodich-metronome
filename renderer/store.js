import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {hashHistory} from 'react-router';
import {routerMiddleware} from 'react-router-redux';
import rootReducer from './reducers/index.js';

const router = routerMiddleware(hashHistory),
    enhancer = applyMiddleware(thunk, router);

export default function configureStore(initialState) {
    return createStore(rootReducer, initialState, enhancer);
}
