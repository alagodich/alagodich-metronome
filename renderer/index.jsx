/* eslint no-console: 0 */

import React from 'react';
import {render} from 'react-dom';
import {Provider} from 'react-redux';
import {syncHistoryWithStore} from 'react-router-redux';
import {ipcRenderer} from 'electron';
import {Router, Route, hashHistory, IndexRoute} from 'react-router';
import configureStore from './store';

// import MainWindow from './components/MainWindow.jsx';
import App from './containers/App.jsx';
import HomePage from './containers/HomePage.jsx';
import CounterPage from './containers/CounterPage.jsx';

const routes = (
    <Route path={'/'} component={App}>
        <IndexRoute component={HomePage} />
        <Route path="/counter" component={CounterPage} />
    </Route>
    ),
    store = configureStore(),
    history = syncHistoryWithStore(hashHistory, store);

ipcRenderer.on('store-update', (event, arg) => {
    console.log('store-update recieved', event, arg);
});

ipcRenderer.send('window-connected');

window.onload = () => {
    const container = document.querySelector('.app');
    if (!container) {
        return;
    }
    // render(<MainWindow ipc={ipcRenderer} />, container);
    render(
        <Provider store={store}>
            <Router history={history} routes={routes} />
        </Provider>,
        container
    );
};
