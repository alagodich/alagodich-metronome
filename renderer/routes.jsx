import React from 'react';
import {Route, IndexRoute} from 'react-router';
import App from './containers/App.jsx';
import HomePage from './containers/HomePage.jsx';
import CounterPage from './containers/CounterPage.jsx';
import {metronomePage, onInit as initMetronome} from './containers/metronomePage.jsx';
import VotingPage from './containers/VotingPage.jsx';

// export default dispatch => (
//     <Route path={'/'} component={App}>
//         <IndexRoute component={HomePage} />
//         <Route path="/counter" component={CounterPage} />
//         <Route path="/metronome" component={metronomePage()} onEnter={initMetronome(dispatch)} />
//         <Route path="/voting" component={VotingPage} />
//     </Route>
// );
export default dispatch => (
    <Route path={'/'} component={App}>
        <IndexRoute component={metronomePage()} onEnter={initMetronome(dispatch)} />
    </Route>
);
