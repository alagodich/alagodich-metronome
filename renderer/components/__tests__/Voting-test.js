import React from 'react';
import Voting from '../Voting';
import renderer from 'react-test-renderer';

jest.mock('electron', () => ({
    ipcRenderer: ({
        on: jest.fn(() => {})
    })
}));

test('voting', () => {
    const vote = () => {},
        pair = ['Trainspotting', '28 Days Later'],
        component = renderer.create(
            <Voting pair={pair} vote={vote} />
        ),
        tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('voted', () => {
    const vote = () => {},
        pair = ['Trainspotting', '28 Days Later'],
        component = renderer.create(
            <Voting pair={pair} vote={vote} hasVoted={pair[0]}/>
        ),
        tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('winner', () => {
    const pair = ['Trainspotting', '28 Days Later'],
        component = renderer.create(
            <Voting pair={pair} winner={pair[0]} />
        ),
        tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
