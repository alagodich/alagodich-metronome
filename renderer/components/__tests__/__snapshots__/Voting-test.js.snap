exports[`test voted 1`] = `
<div
  className="voting">
  <div
    className="voting ui buttons">
    <button
      className="ui button"
      onClick={undefined}>
      <i
        className="heart icon" />
      Trainspotting
    </button>
    <button
      className="ui button"
      onClick={undefined}>
      28 Days Later
    </button>
  </div>
</div>
`;

exports[`test voting 1`] = `
<div
  className="voting">
  <div
    className="voting ui buttons">
    <button
      className="ui button"
      onClick={undefined}>
      Trainspotting
    </button>
    <button
      className="ui button"
      onClick={undefined}>
      28 Days Later
    </button>
  </div>
</div>
`;

exports[`test winner 1`] = `
<div
  className="voting">
  <span>
    Trainspotting
     winner!
  </span>
</div>
`;
